### PR TITLE
Remove PropertyGrid remoting code

### PR DESCRIPTION
**Pull request by hughbe**
This is not supported in .NET Core

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/winforms/pull/3422)

### CODE REVIEW

File: src/System.Windows.Forms/src/System/Windows/Forms/PropertyGrid.cs [hunk 1]
```diff
@@ -5285,24 +5281,6 @@ public void RemoveTabType(Type propertyTabType)
...
-        ///  even for objects that DON'T actually implement the interface.  The error here
-        ///  is raised later when you make your first call on that interface pointer:  you
-        ///  get a remoting exception.
-        ///
-        ///  This is a big problem for code that does "is" and "as" checks to detect the
-        ///  presence of an interface.  We do that all over the place here, so we do a check
-        ///  during parameter validation to see if an object implements IUnimplemented.  If it
-        ///  does, we know that what we really have is a lying remoting proxy, and we bail.
-        /// </summary>
-        private interface IUnimplemented { }
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/PropertyGrid.cs:5304]: 😱

**Comment by RussKie** [comment]
@dreddy-work @Tanya-Solyanik this won't cause any dramas for the Designer, will it?

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ControlTests.Handlers.cs [hunk 2]
```diff
@@ -3375,11 +3381,13 @@ public void Control_OnMouseCaptureChanged_Invoke_CallsMouseCaptureChanged(EventA
             control.MouseCaptureChanged += handler;
             control.OnMouseCaptureChanged(eventArgs);
             Assert.Equal(1, callCount);
+            Assert.False(control.IsHandleCreated);
 
            // Remove handler.
            control.MouseCaptureChanged -= handler;
            control.OnMouseCaptureChanged(eventArgs);
            Assert.Equal(1, callCount);
+            Assert.False(control.IsHandleCreated);
```
[the remark this answers is not in the document]
Comment (hughbe) [reply src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ControlTests.Handlers.cs:3390]: looks like the problem was the indentation of the code above. Fixed

**Comment by JeremyKuhne** [comment]
.NET Remoting isn't supported on .NET Core so removing this should be fine. Sort of an icky way to check that you have a proxy, but I have no idea if there were alternatives.

**Pull request merged**